### ISSUE TITLE
Top-right icons change to debugger icons on non-debugger tab

### ISSUE THREAD

**Issue by tristanbob** [4ian/GDevelop#3296]
## Describe the bug

Tested on version 5.0.122

https://user-images.githubusercontent.com/8879811/143534314-d0d6e50b-b653-48ca-bf6b-40c60d8fc114.mp4

## To Reproduce
1. Preview a game
2. Open the Debugger
3. Close preview but leave Debugger tab open
4. Change to a non-Debugger tab
5. Preview a game
6. The top-right icons change to the debugger icons (play/pause/etc)

**Comment by ClementPasteau** [comment]
I've looked briefly but couldn't think of a straightforward solution.
Basically, the function `updateToolbar` of the debugger is called after the function of the eventsheet, causing it to be replaced. both calls are not synchronised so I assume the behavior can be random.

They are both called in multiple events and the setToolbar of the Debugger is not aware of the context of the current tab opened.

https://github.com/4ian/GDevelop/blob/34cbcdbc3a02e340580b31cb4cab8d507391719d/newIDE/app/src/Debugger/index.js#L81-L86

https://github.com/4ian/GDevelop/blob/34cbcdbc3a02e340580b31cb4cab8d507391719d/newIDE/app/src/EventsSheet/index.js#L291-L297

**Comment by 4ian** [comment]
Weird still because the `this.props.isActive` should bail out if the tab is not active (i.e: focused). Do you want to check this assumption by logging it?

**Comment by ClementPasteau** [comment]
I've logged it, and `isActive` is still true. This is where I probably misunderstood the prop `isActive` as I thought it meant "is the debugger open" :)
I'll dig a bit more to understand why it doesn't switch to false then

**Comment by 4ian** [comment]
MainFrame has at least logic to set it active only if the tab is focused, but maybe there is somewhere a shouldComponentUpdate or some other kind of optimization making the boolean staying at true 🤔

![image](https://user-images.githubusercontent.com/1280130/144296968-5d3aebbf-a0e9-4c13-b81c-826736e2d81c.png)

**Comment by ClementPasteau** [comment]
Found the culprit I think: https://github.com/4ian/GDevelop/commit/bc27364bb875b016df5486a8aa1e55dd172098d3#diff-f0067651e7857e1f7c24468e87cf9477614cd8a26a8eb9002762c2a63ccefd8dR26-R29

Quite old as well.
Debugger is not rendered again if not focused. which kinda makes sense, but then leaves the callbacks registered in the debugger.

**Comment by 4ian** [comment]
Yes I would keep this (otherwise a re-render of the app, which happens fairly frequently like if you open the project manager). But instead of doing the `if (!this.props.isActive) return;` in `Debugger`, we should do it in the `updateToolbar()` of `DebuggerEditorContainer`, i.e here: https://github.com/4ian/GDevelop/commit/bc27364bb875b016df5486a8aa1e55dd172098d3#diff-f0067651e7857e1f7c24468e87cf9477614cd8a26a8eb9002762c2a63ccefd8dR40


Basically, we should not pass `isActive` to children, because the "container" (DebuggerEditorContainer, possibly other from the other editors) will stop the updates if it's not active anyway.

**Comment by ClementPasteau** [comment]
I don't think this is enough.
The problem comes from the `onConnectionOpened` of the Debugger, the callback is registered and will be triggered no matter if the container has passed props or not because it has been rendered once.

https://github.com/4ian/GDevelop/blob/106549e5fa1dc92ec6fa76f0a9dcdcef71a17784/newIDE/app/src/Debugger/index.js#L183-L191

+ the `componentWillReceiveProps` (which is supposed to unregister the callbacks) isn't triggered as expected because the top component has decided to not re-render it.

It means the debugger toolbar will always be rendered when the preview starts, no matter where you are in the app, if you have opened the debugger once.

I feel that registering callbacks & controlling a shouldComponentUpdate in a component above is dangerous.
I've created a fix in #3324 

Also, I don't understand what the `updateToolbar()` function is for? It doesn't seem to be used
